### ISSUE TITLE
Implement the `run:` step

### ISSUE THREAD

**Issue by aoldershaw** [concourse/concourse#7061]
## Summary

Allow users to invoke messages on `prototypes` using the `run:` step, e.g.:

```yaml
prototypes:
- name: some-prototype
  ...

jobs:
- name: job
  plan:
  - run: some-message
    type: some-prototype
    params:
      foo: bar
```

There is an open question about how to provide inputs to prototypes, and how prototypes can register outputs. Discussion in concourse/rfcs#103

## Context

* RFC: concourse/rfcs#37 and concourse/rfcs#103
* Depends on #7060 
* Part of #6905

**Comment by taylorsilva** [comment]
# Issue Bankruptcy

This issue is being closed as part of 2025's **Issue Bankruptcy** declaration. [See here for details](https://blog.concourse-ci.org/posts/2025-01-06-concourse-2024-review/)

If you think this issue still has merit then feel free to open a new issue and reference this one.
